src/domain/orders/details/create-fulfillment/index.tsx

```diff
@@ -0,0 +1,314 @@
+
+  const items =
+    "items" in orderToFulfill
+      ? orderToFulfill.items
+      : orderToFulfill.additional_items
+
+  const createOrderFulfillment = useAdminCreateFulfillment(orderId)
+  const createSwapFulfillment = useAdminFulfillSwap(orderId)
+  const createClaimFulfillment = useAdminFulfillClaim(orderId)
+
+  const toaster = useToaster()
+
+  const createFulfillment = () => {
+    const [type] = orderToFulfill.id.split("_")
+
+    type actionType =
+      | typeof createOrderFulfillment
+      | typeof createSwapFulfillment
+      | typeof createClaimFulfillment
+
+    let action: actionType = createOrderFulfillment
+    let successText = "Successfully fulfilled order"
+    let requestObj
+
+    const preparedMetadata = metadata.reduce((acc, next) => {
+      if (next.key) {
+        return {
+          ...acc,
+          [next.key]: next.value,
+        }
+      } else {
+        return acc
+      }
+    }, {})
+
+    switch (type) {
+      case "swap":
+        action = createSwapFulfillment
+        successText = "Successfully fulfilled swap"
+        requestObj = {
+          metadata: preparedMetadata,
+          no_notification: noNotis,
+        } as AdminPostOrdersOrderSwapsSwapFulfillmentsReq
+        break
+
+      case "claim":
+        action = createClaimFulfillment
+        successText = "Successfully fulfilled claim"
+        requestObj = {
+          metadata: preparedMetadata,
+          no_notification: noNotis,
+        } as AdminPostOrdersOrderClaimsClaimFulfillmentsReq
+        break
+
+      default:
+        requestObj = {
+          metadata: preparedMetadata,
+          no_notification: noNotis,
+        } as AdminPostOrdersOrderFulfillmentsReq
+        requestObj.items = toFulfill
+          .map((itemId) => ({ item_id: itemId, quantity: quantities[itemId] }))
+          .filter((t) => !!t)
+        break
+    }
+
+    action.mutate(requestObj, {
+      onSuccess: () => {
+        toaster(successText, "success")
+        handleCancel()
+      },
+      onError: (err) => toaster(getErrorMessage(err), "error"),
+    })
+  }
+
+  return (
+    <Modal handleClose={handleCancel}>
+      <Modal.Body>
+        <Modal.Header handleClose={handleCancel}>
+          <span className="inter-xlarge-semibold">Create Fulfillment</span>
+        </Modal.Header>
+        <Modal.Content>
+          <div className="flex flex-col">
+            <span className="inter-base-semibold mb-2">Items</span>
+            <ItemsTable
+              items={items}
+              toFulfill={toFulfill}
+              setToFulfill={setToFulfill}
+              quantities={quantities}
+              setQuantities={setQuantities}
+            />
+            <div className="mt-4">
+              <Metadata metadata={metadata} setMetadata={setMetadata} />
+            </div>
+          </div>
+        </Modal.Content>
+        <Modal.Footer>
+          <div className="flex w-full h-8 justify-end">
+            <Button
+              variant="ghost"
```

Comment:
missing the `no notificaitons` checkbox 

design: 
![image](https://user-images.githubusercontent.com/88927411/152490794-492f6acd-4351-4be5-9d6b-d48280e6533b.png)
```suggestion
        <Modal.Footer>
          <div className="flex w-full h-8 justify-end">
           <div
              className="items-center h-full flex cursor-pointer"
              onClick={() => setNoNotis(!noNotis)}
            >
              <div
                className={`w-5 h-5 flex justify-center text-grey-0 border-grey-30 border rounded-base ${
                  !noNotis && "bg-violet-60"
                }`}
              >
                <span className="self-center">
                  {!noNotis && <CheckIcon size={16} />}
                </span>
              </div>
              <input
                id="noNotification"
                className="hidden"
                name="noNotification"
                checked={!noNotis}
                type="checkbox"
              />
              <span className="ml-3 flex items-center text-grey-90 gap-x-xsmall">
                Send notifications
                <InfoTooltip content="" />
              </span>
            </div>
            <Button
              variant="ghost"
```

Reply:
still needs a `justify-between` and wrapper for the buttons

Reply:
Thanks @pKorsholm ! Completely missed this. Can I get you to give another review now? 👍